### ISSUE TITLE
mock_dynamodb2 cannot delete attribute as part of `update_item`

### ISSUE THREAD

**Issue by stpierre** [spulec/moto#3797]
## Reporting Bugs

In moto 2.0.2, it's no longer possible to delete an attribute with `AttributeUpdates` as part of an `update_item()` call. For instance:

```python
import boto3
import moto

with moto.mock_dynamodb2():
    name = "TestTable"
    conn = boto3.client(
        "dynamodb",
        region_name="eu-west-2",
        aws_access_key_id="ak",
        aws_secret_access_key="sk")
    conn.create_table(
        TableName=name,
        AttributeDefinitions=[{
            'AttributeName': 'name',
            'AttributeType': 'S'
        }],
        KeySchema=[{
            'AttributeName': 'name',
            'KeyType': 'HASH'
        }],
    )

    item_name = 'foo'
    conn.put_item(
        TableName=name, Item={
            'name': {
                'S': item_name
            },
            'extra': {
                'S': 'bar'
            }
        })

    conn.update_item(
        TableName=name,
        Key={'name': {
            'S': item_name
        }},
        AttributeUpdates={'extra': {
            'Action': 'DELETE'
        }})
```

This fails with:

```
Traceback (most recent call last):
  File "./test.py", line 34, in <module>
    conn.update_item(
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/client.py", line 662, in _make_api_call
    http, parsed_response = self._make_request(
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/client.py", line 682, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/endpoint.py", line 102, in make_request
    return self._send_request(request_dict, operation_model)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/endpoint.py", line 136, in _send_request
    while self._needs_retry(attempts, operation_model, request_dict,
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/endpoint.py", line 253, in _needs_retry
    responses = self._event_emitter.emit(
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/hooks.py", line 356, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/hooks.py", line 228, in emit
    return self._emit(event_name, kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/hooks.py", line 211, in _emit
    response = handler(**kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/retryhandler.py", line 183, in __call__
    if self._checker(attempts, response, caught_exception):
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/retryhandler.py", line 250, in __call__
    should_retry = self._should_retry(attempt_number, response,
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/retryhandler.py", line 269, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/retryhandler.py", line 316, in __call__
    checker_response = checker(attempt_number, response,
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/retryhandler.py", line 222, in __call__
    return self._check_caught_exception(
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
    raise caught_exception
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/endpoint.py", line 197, in _do_get_response
    responses = self._event_emitter.emit(event_name, request=request)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/hooks.py", line 356, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/hooks.py", line 228, in emit
    return self._emit(event_name, kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/botocore/hooks.py", line 211, in _emit
    response = handler(**kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/core/models.py", line 363, in __call__
    status, headers, body = response_callback(
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/core/responses.py", line 202, in dispatch
    return cls()._dispatch(*args, **kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/core/responses.py", line 312, in _dispatch
    return self.call_action()
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/core/utils.py", line 259, in _wrapper
    response = f(*args, **kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/core/utils.py", line 291, in _wrapper
    response = f(*args, **kwargs)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/dynamodb2/responses.py", line 90, in call_action
    response = getattr(self, endpoint)()
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/dynamodb2/responses.py", line 776, in update_item
    item = self.dynamodb_backend.update_item(
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/dynamodb2/models/__init__.py", line 1309, in update_item
    item.validate_no_empty_key_values(attribute_updates, table.key_attributes)
  File "/home/stpierre/moto-bug-demo/lib/python3.8/site-packages/moto/dynamodb2/models/__init__.py", line 114, in validate_no_empty_key_values
    new_value = next(iter(update_action["Value"].values()))
KeyError: 'Value'
```

Supplying a bogus placeholder `Value` key doesn't work, either; if you supply any value other than a set (`SS`) you get an error that you can't ADD a value; if you supply a set, it thinks it's supposed to delete the supplied values from the existing set, or failing that, set the value to a set. For instance, if you do:

```python
    conn.update_item(
        TableName=name,
        Key={'name': {
            'S': item_name
        }},
        AttributeUpdates={'extra': {
            'Value': {'SS': []},
            'Action': 'DELETE'
        }})
```

...then after the update, the value of the `extra` key will be the empty set.

**Comment by bblommers** [comment]
Thanks for raising this @stpierre - marking it as a bug

**Comment by hhschu** [comment]
Is there any update on this issue?

**Comment by bblommers** [comment]
This is now fixed as of moto >= 2.0.11.dev2
